lib/Extractor/KLEEBuilder.cpp

```diff
@@ -136,7 +140,24 @@ ref<Expr> ExprBuilder::makeSizedArrayRead(unsigned Width, StringRef Name,
       new Array(ArrayNames.makeName(NameStr), 1, 0, 0, Expr::Int32, Width));
   ArrayVars.push_back(Origin);
 
+  std::vector<unsigned> ZeroBits, OneBits;
   UpdateList UL(Arrays.back().get(), 0);
+  if (Origin && Origin->K == Inst::Var && (Origin->KnownZeros != APInt(Width, 0) ||
+                                           Origin->KnownOnes != APInt(Width, 0))) {
+    ref<Expr> KnownZerosExpr = klee::ConstantExpr::create(0, Expr::Bool);
```

Comment:
there's a better way to test an APInt for zero, see APInt.h. constructing fresh zero-valued APInts is not elegant or efficient.